tinker/faculty_bios/faculty_bio_controller.py

```diff
@@ -122,21 +123,121 @@ def _iterate_child_xml(self, child, author):
+        max_jobs = 0
+        for jobs in child.iterfind('.//job-titles'):
+            max_jobs += 1
+            school = self.get_faculty_data(jobs, 'school')
+            page_values['school' + str(max_jobs)] = school
+            page_values['job_title' + str(max_jobs)] = self.get_faculty_data(jobs, 'job_title')
+            page_values['department' + str(max_jobs)] = ""
+            if school == 'College of Arts and Sciences':
+                page_values['department' + str(max_jobs)] = self.get_faculty_data(jobs, 'department')
+                if self.get_faculty_data(jobs, 'department-chair') == 'Yes':
+                    page_values['job_title' + str(max_jobs)] = 'Department Chair'
+            elif school == 'College of Adult and Professional Studies':
+                page_values['department' + str(max_jobs)] = self.get_faculty_data(jobs, 'adult-undergrad-program')
+                if self.get_faculty_data(jobs, 'program-director') == 'Yes':
+                    page_values['job_title' + str(max_jobs)] = 'Program Director'
+            elif school == 'Graduate School':
+                page_values['department' + str(max_jobs)] = self.get_faculty_data(jobs, 'graduate-program')
+                if self.get_faculty_data(jobs, 'program-director') == 'Yes':
+                    page_values['job_title' + str(max_jobs)] = 'Program Director'
+            elif school == 'Bethel Seminary':
+                page_values['department' + str(max_jobs)] = self.get_faculty_data(jobs, 'seminary-program')
+                if self.get_faculty_data(jobs, 'program-director') == "Yes":
+                    page_values['job_title' + str(max_jobs)] = 'Program Director'
+                elif self.get_faculty_data(jobs, 'lead-faculty') != "Other":
+                    page_values['job_title' + str(max_jobs)] = jobs.find('lead-faculty').text
+
+        page_values['max-jobs'] = max_jobs
+
+        # Iterates through all the education fields and adds it to the dictionary on the fly
+        max_edu = 0
+        for edu in child.iterfind('.//education'):
+            max_edu += 1
+            page_values['school-edu' + str(max_edu)] = unidecode(self.get_faculty_data(edu, 'school'))
+            page_values['degree-earned' + str(max_edu)] = self.get_faculty_data(edu, 'degree-earned')
+            page_values['year' + str(max_edu)] = self.get_faculty_data(edu, 'year')
+
+        page_values['max-edu'] = max_edu
+
+        # return the dictionary
+        return page_values
+
+    # A method used to extract all the inner text from the wysiwyg's
+    def wysiwyg_inner_text(self, child, path_tail):
+        string = ""
+        for information in child.find('.//' + str(path_tail)).itertext():
+            string += information
+            string = string.rstrip()
+            string += '\n'
+        string = string.lstrip()
+        return string.rstrip()
+
+    def get_faculty_data(self, child, path_tail):
```

Comment:
these 2 methods are very smooth 👍 

Might be worth a comment, just cause

Reply:
👍